examples/OAuthConsumer/Program.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright 2022 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+/// <summary>
+///     An example showing consumer 
+///     with a custom OAUTHBEARER token implementation.
+/// </summary>
+namespace Confluent.Kafka.Examples.OAuthConsumer
+{
+    /// <summary>
+    ///     A class to store the token and related properties.
+    /// </summary>
+    class OAuthBearerToken
+    {
+        public string TokenValue { get; set; }
+        public long Expiration { get; set; }
+        public String Principal { get; set; }
+        public Dictionary<String, String> Extensions { get; set; }
+    }
+
+    public class Program
+    {
+        private const String OauthConfigRegexPattern = "^(\\s*(\\w+)\\s*=\\s*(\\w+))+\\s*$"; // 1 or more name=value pairs with optional ignored whitespace
+        private const String OauthconfigKeyValueRegexPattern = "(\\w+)\\s*=\\s*(\\w+)"; // Extract key=value pairs from OAuth Config

```

Comment:
Inconsistent casing, should be OauthConfigKey...

Reply:
Thanks, updated.